### PR TITLE
Adjust pytest pin to allow Python 3.11 build

### PR DESCRIPTION
**Pull request by corydolphin**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

```
dlib (2) Error: not solvable ([bot CI job](https://github.com/regro/autotick-bot/actions/runs/3581275141)): main: ['linux_64_numpy1.23python3.11.____cpython: Encountered problems while solving:\n - package pytest-7.0.1-py37h89c1867_0 requires python_abi 3.7.* *_cp37m, but none of the providers can be installed\n', 'osx_64_numpy1.23python3.11.____cpython: Encountered problems while solving:\n - package pytest-7.0.1-py37hf985489_0 requires python_abi 3.7.* *_cp37m, but none of the providers can be installed\n', 'osx_arm64_numpy1.23python3.11.____cpython: Encountered problems while solving:\n - package pytest-7.0.1-py38h10201cd_0 requires python_abi 3.8.* *_cp38, but none of the providers can be installed\n', 'win_64_numpy1.23python3.11.____cpython: Encountered problems while solving:\n - package pytest-7.0.1-py37h03978a9_0 requires python_abi 3.7.* *_cp37m, but none of the providers can be installed\n']
Immediate Children: menpodetect, menpofit
```

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Interesting, I haven't seen this error in the CI pipeline before (on win):
```bash
Traceback (most recent call last):
  File "C:\Miniforge\Scripts\conda-mambabuild-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 256, in main
    call_conda_build(action, config)
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 228, in call_conda_build
    result = api.build(
  File "C:\Miniforge\lib\site-packages\conda_build\api.py", line 186, in build
    return build_tree(
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 3091, in build_tree
    packages_from_this = build(metadata, stats,
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 2374, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 1694, in bundle_conda
    conda_package_handling.api.create(
  File "C:\Miniforge\lib\site-packages\conda_package_handling\api.py", line 117, in create
    raise err
  File "C:\Miniforge\lib\site-packages\conda_package_handling\api.py", line 110, in create
    out = format.create(prefix, file_list, out_fn, out_folder, **kw)
  File "C:\Miniforge\lib\site-packages\conda_package_handling\conda_fmt.py", line 111, in create
    component_stream.close()
zstd.ZstdError: zstd compress error: Allocation error : not enough memory
```

**Comment by corydolphin** [comment]
> Interesting, I haven't seen this error in the CI pipeline before (on win):
> 
> ```python
> Traceback (most recent call last):
>   File "C:\Miniforge\Scripts\conda-mambabuild-script.py", line 9, in <module>
>     sys.exit(main())
>   File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 256, in main
>     call_conda_build(action, config)
>   File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 228, in call_conda_build
>     result = api.build(
>   File "C:\Miniforge\lib\site-packages\conda_build\api.py", line 186, in build
>     return build_tree(
>   File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 3091, in build_tree
>     packages_from_this = build(metadata, stats,
>   File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 2374, in build
>     newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
>   File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 1694, in bundle_conda
>     conda_package_handling.api.create(
>   File "C:\Miniforge\lib\site-packages\conda_package_handling\api.py", line 117, in create
>     raise err
>   File "C:\Miniforge\lib\site-packages\conda_package_handling\api.py", line 110, in create
>     out = format.create(prefix, file_list, out_fn, out_folder, **kw)
>   File "C:\Miniforge\lib\site-packages\conda_package_handling\conda_fmt.py", line 111, in create
>     component_stream.close()
> zstd.ZstdError: zstd compress error: Allocation error : not enough memory
> ```

It sounds like there is a thread on gitter about this. It appears to be transient, and I hit it in a different build earlier. https://gitter.im/conda-forge/conda-forge.github.io?at=6387c05c426f9311ffcc0bd6

**Comment by corydolphin** [comment]
@conda-forge-admin please rerender

**Comment by Tobias-Fischer** [comment]
LGTM

File: conda-forge.yml [hunk 1]
```diff
@@ -7,5 +7,3 @@ test_on_native_only: true
 github:
   branch_name: main
   tooling_branch_name: main
-conda_build:
```
Comment (corydolphin) [inline conda-forge.yml:10]: image

**Comment by corydolphin** [comment]
> LGTM

Thank you for the review! I made one subsequent change to hopefully fix the windows build issue. It looks like I do not have merge permissions. Are you able to merge (once CI passes 🤞 ).

**Pull request merged**